src/react-components/src/components/01-atoms/Forms/TextField.tsx

```diff
@@ -14,22 +14,22 @@ export interface TextFieldProps {
   onChange?: (event: React.FormEvent) => void;
 }
 export default function TextField(props: TextFieldProps) {
-  const { labelId, ariaLabel, isRequired, blockName, modifiers, value, placeholderText, onChange } = props;
+  const { ariaLabel, ariaLabelledBy, isRequired, blockName, modifiers, value, placeholderText, onChange } = props;
   let textfieldBlockName = blockName ? blockName : "form-item";
 
+  if (!ariaLabel && !ariaLabelledBy) {
+    throw new Error("Either label or ariaLabel must be defined");
+  }
+
   let textFieldProps = {
     className: bem("input", modifiers, textfieldBlockName),
     "type": "text",
     value: value,
     onChange: onChange,
+    "aria-label": !ariaLabelledBy ? ariaLabel : undefined,
+    "aria-labelledby": ariaLabelledBy
```

Comment:
You can include `aria-required` here if you don't want the extra `if (isRequired)` check. Maybe do something like `"aria-required": isRequired`, make `isRequired` optional, and assign a default prop value of `false` (?) It makes sense to not make it required unless if it's necessary and it could help reduce props when calling it - `<TextField ariaLabel="text field" onChange={() => {}}/>;` instead of `<TextField ariaLabel="text field" isRequired={false} onChange={() => {}}/>;`
But this is minor.

Reply:
`TextField` should rarely (almost never) be used on its own, so I like the idea of forcing people to pass in isRequired (because it really should be other DS components calling `TextField`), and those DS components should know whether or not it's required.

Reply:
Maybe there should be a subset of "helper" components (a folder for them) that only DS uses  so anyone reading the source doesn't think that it's a component they can use.

Reply:
Agreed, though this should probably be reflected in Metronome as well (components that should only be used by other components).  Yet another thing to go on the pile!